Escape Message Preview Links At Any Position In The Message Content

BlackBearFTW opened the pull request:
This bugfix makes sure the discord message refrence url escape characters can be used anywhere.

Also `!link` has been changed to `<link>` to keep the discord bot consistent with discords methods

BlackBearFTW committed:
Update DiscordMessageLinkHandler.ts

BlackBearFTW commented:
Yeah I still have to write tests

BlackBearFTW committed:
Wrote tests and made channel clickable

- Test added to check if <link> gets escaped if message starts and ends with <>
- Test added to check if <link> gets escaped mid sentence
- Made channel clickable in preview

BlackBearFTW commented:
Tests have been written, all done now. 

Thank you for the collab, hope ill get to do some more in the future!

LamboCreeper commented on src/event/handlers/DiscordMessageLinkHandler.ts:
```diff
@@ -11,10 +11,15 @@ class DiscordMessageLinkHandler extends EventHandler {
 		if (message.content.match(messageRegex)) {
 			const linkIndex = message.content.search(messageRegex);
+
+			// Gets the position of the url from a string
```
Useless comment, code is self explanatory!

BlackBearFTW replied:
Removed 👍

LamboCreeper reviewed (approved):
Looks good! Thanks 🙂

BlackBearFTW committed:
Update DiscordMessageLinkHandler.ts

BlackBearFTW commented:
I removed the comment!

BlackBearFTW committed:
changed coverage for Lambo and improved code

- Lambo asked me to change coverage.yml
- Now it will return when linkIndex is -1

Jason2605 commented on src/event/handlers/DiscordMessageLinkHandler.ts:
```diff
@@ -8,15 +8,20 @@ class DiscordMessageLinkHandler extends EventHandler {
-
 		if (message.content.match(messageRegex)) {
 			const linkIndex = message.content.search(messageRegex);
+			if (linkIndex == -1) return;
```
This is the change i was suggesting, there's no need for `.match` as `.search` returns -1 if there was no match.

```suggestion
		const linkIndex = message.content.search(messageRegex);
		
		if (linkIndex !== -1) {
```

BlackBearFTW committed:
Changed to match Jason's suggestion

BlackBearFTW committed:
Update DiscordMessageLinkHandler.ts

BlackBearFTW committed:
Update DiscordMessageLinkHandler.ts

BlackBearFTW committed:
Trying to fix the linter issues
